Include/Rover/TrialView.hpp

```diff
@@ -81,20 +81,20 @@ namespace Rover {
 
           //! Returns a reference to the Sample the iterator would point to
           //! if it was advanced by the offset.
-          const Sample& operator [](std::ptrdiff_t offset) const;
+          Sample operator [](std::ptrdiff_t offset) const;
```

Comment:
Why is this returning by value instead of reference?

Reply:
Frankly, nobody is typically aware of this operator of the RandomAccessIterator concept, and it is hard for me to imagine how it is properly used. I can indeed make it return a reference by storing the sample at the offset in m_sample as well, no additional overhead imposed.

Reply:
Return the reference by calling ```m_get[offset]```. ```m_get``` returns a ```const&```.